crewai_tools/tools/couchbase_tool/couchbase_tool.py

```diff
@@ -0,0 +1,254 @@
+            except Exception as e:
+                raise ValueError(
+                    "Error connecting to couchbase. "
+                    "Please check the connection and credentials"
+                ) from e
+
+            # check if bucket exists
+            if not self._check_bucket_exists():
+                raise ValueError(
+                    f"Bucket {self.bucket_name} does not exist. "
+                    " Please create the bucket before searching."
+                )
+
+            self._check_scope_and_collection_exists()
+            self._check_index_exists()
+        else:
+            import click
+
+            if click.confirm(
+                "The 'couchbase' package is required to use the CouchbaseFTSVectorSearchTool. "
+                "Would you like to install it?"
+            ):
+                import subprocess
+
+                subprocess.run(["uv", "add", "couchbase"], check=True)
+            else:
+                raise ImportError(
+                    "The 'couchbase' package is required to use the CouchbaseFTSVectorSearchTool. "
+                    "Please install it with: uv add couchbase"
+                )
+
+    def _run(self, query: str) -> str:
+        """Execute a vector search query against the Couchbase index.
+
+        Args:
+            query: The search query string.
+
+        Returns:
+            A JSON string containing the search results.
+
+        Raises:
+            ImportError: If the 'couchbase' package is not installed.
+            ValueError: If the search query fails or returns results without fields.
+        """
+        if not COUCHBASE_AVAILABLE:
+            raise ImportError(
+                "You are missing the 'couchbase' package. Would you like to install it?"
+            )
+
+        query_embedding = self.embedding_function(query)
+        fields = ["*"]
+
+        search_req = search.SearchRequest.create(
+            VectorSearch.from_vector_query(
+                VectorQuery(
+                    self.embedding_key,
+                    query_embedding,
+                    self.limit
+                )
+            )
+        )
+
+        try:
+            if self.scoped_index:
+                search_iter = self._scope.search(
+                    self.index_name,
+                    search_req,
+                    SearchOptions(
+                        limit=self.limit,
+                        fields=fields,
+                    )
+                )
+            else:
+                search_iter = self.cluster.search(
+                    self.index_name,
+                    search_req,
+                    SearchOptions(
+                        limit=self.limit,
+                        fields=fields
+                    )
+                )
+
+            json_response = ""
+
+            for row in search_iter.rows():
+                if row.fields:
+                    json_response += json.dumps(row.fields, indent=2)
```

Comment:
QQ:
You are not returning a valid JSON here. Is that expected? What about make it instead

```
output = []

for ...
    output.append(row.fieds)

return json.dumps(output, ...)


```